cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -449,7 +460,13 @@ public OptimizerResult optimizations(ClusterModel clusterModel,
       LOG.trace("Broker level stats after optimization: {}%n", clusterModel.brokerStats(null));
     }
 
-    Set<ExecutionProposal> proposals = AnalyzerUtils.getDiff(initReplicaDistribution, initLeaderDistribution, clusterModel);
+    Set<ExecutionProposal> proposals =
+        AnalyzerUtils.getDiff(initReplicaDistributionForProposalGeneration != null ? initReplicaDistributionForProposalGeneration
```

Comment:
The difference between `initReplicaDistributionForProposalGeneration` and `initReplicaDistribution` is confusing. Can we name them more clear (ideally the parameter name)?

Reply:
Add more comments in JavaDoc to explain what it is used for.